validator/tests/test_scheduler/yaml_scheduler_tester.py

```diff
@@ -191,8 +192,14 @@ def run_scheduler(self,
                     txns_to_process.append(txn_info)
                 else:
                     stop = True
-            t_info = txns_to_process.pop()
+
+            if txns_executed_fifo:
```

Comment:
I will switch to a deque. The size of the list would depend on the number of transactions in the yaml file, and the number of independent transactions at any one time for the parallel scheduler. When the serial scheduler is being tested, txns_to_process is always 1.